src/Raven.Server/Documents/ShardedHandlers/ShardedSmugglerHandler.cs

```diff
@@ -0,0 +1,182 @@
+{
+    public class ShardedSmugglerHandler : ShardedRequestHandler
+    {
+        [RavenShardedAction("/databases/*/smuggler/export", "POST")]
+        public async Task PostExport()
+        {
+            var result = new SmugglerResult();
+
+            using (ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            {
+                var operationId = GetLongQueryString("operationId", false) ?? ServerStore.Operations.GetNextOperationId();
+
+                try
+                {
+                    await Export(context, ShardedContext.DatabaseName, (options, blittableOptions, startDocumentEtag, startRaftIndex, onProgress, _, token) =>
+                            ExportShardedDatabaseInternalAsync(options, blittableOptions, startDocumentEtag, startRaftIndex, onProgress, context, token),
+                        ServerStore.Operations, operationId);
+                }
+                catch (Exception e)
+                {
+                    if (Logger.IsOperationsEnabled)
+                        Logger.Operations("Export failed .", e);
+
+                    result.AddError($"Error occurred during export. Exception: {e.Message}");
+                    await WriteResultAsync(context, result, ResponseBodyStream());
+
+                    HttpContext.Abort();
+                }
+            }
+        }
+
+        public async Task<IOperationResult> ExportShardedDatabaseInternalAsync(
+            DatabaseSmugglerOptionsServerSide options,
+            BlittableJsonReaderObject blittableJson,
+            long startDocumentEtag,
+            long startRaftIndex,
+            Action<IOperationProgress> onProgress,
+            JsonOperationContext jsonOperationContext,
+            OperationCancelToken token)
+        {
+            var tcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            var oldOperateOnType = options.OperateOnTypes;
+
+            var operateOnTypes = options.OperateOnTypes &= ~(DatabaseItemType.DatabaseRecord |
+                                                             DatabaseItemType.Subscriptions |
+                                                             DatabaseItemType.Identities |
+                                                             DatabaseItemType.Indexes |
+                                                             DatabaseItemType.ReplicationHubCertificates);
+
+            blittableJson = CreateNewOptionBlittableJsonReaderObject(blittableJson, jsonOperationContext, nameof(options.OperateOnTypes), operateOnTypes);
+
+            using (var outputStream = GetOutputStream(ResponseBodyStream(), options))
+            using (var exportOutputStream = new GZipStream(outputStream, CompressionMode.Compress))
+            {
+                await exportOutputStream.WriteAsync(Encoding.UTF8.GetBytes($"{{  \"BuildVersion\" : {ServerVersion.Build}"));
+                for (int i = 0; i < ShardedContext.ShardCount; i++)
```

Comment:
We have to document that export is _not_ consistent across shards, because this can run for hours and have very different state between the nodes.